src/hooks/Transactor.ts

```diff
@@ -181,11 +181,10 @@ export function useTransactor({
           description = JSON.parse(json).message || message
         } catch (_) {
           description = message
-          options?.onError?.(new DOMException(description))
           emitErrorNotification(t`Transaction failed`, { description })
         }
 
-        options?.onDone?.()
+        options?.onError?.(new DOMException(description))
```

Comment:
I am hesitant to change the transactor code as it could have unforeseen consequences elsewhere. We are using it in too many places. Although its not the best practice, I think you are better off using the pattern used elsewhere of detecting the boolean flag that is returned from the transactor call itself.

i.e, 
```ts
const txSuccess = redeemTokensTx(...)
if (!txSuccess) {
  setTransactionPending(false)
  setLoading(false)
}
```

Reply:
FWIW this would be less an issue if we had more test coverage, but I don't think it is worth running this risk right now

Reply:
Good point, is this an essential change for this fix @NakedFool ?

Reply:
@johnnyd-eth It is ser, i will apply right in the morning.

@wraeth-eth that is good point, i was a bit anxious about changing that functionality in there, i will minimize risks with how you explained it to me.

Reply:
Branch is now updated with these changes 🍻